torch/_guards.py

```diff
@@ -175,3 +177,53 @@ def set_export_info(self, guard_type, guarded_class, code_list, obj_weakref):
             None,
         ), "Guarded object must be identical, or None"
         self.obj_weakref = obj_weakref
+
+
+T = TypeVar("T")
+
+
+class Checkpointable(ABC):
+    def copy_graphstate(self) -> Generic[T]:
```

Comment:
nit: why not call this `checkpoint` and `restore` copying is one possible impl, ideally we shouldn't be copying and using generations

Reply:
(1) We copy all over the place in the other copy/restore functions

(2) I don't want to rename for the sake of renaming 

(3) patch this down and run the added test without copy, it fails (as expected)

Reply:
Unless you mean something like a map of generation:guards, in which case I'd rather stay consistent with the upstream generation system. Either we rewrite everything to work that way, and remove all the copies upstream, or we keep it all consistent.

Reply:
yeah, this is a preexisting condition lol